app/auth/views/suppliers.py

```diff
@@ -157,10 +159,108 @@ def supplier_service_prices(service_type_id, category_id=None):
                 ServiceTypePrice.is_current_price)\
         .all()
 
-    return jsonify(prices=[dict(
-        id=p.id,
-        region=dict(state=p.region.state, name=p.region.name),
-        price=format_price(p.price),
-        capPrice=format_price(None if p.service_type_price_ceiling is None else p.service_type_price_ceiling.price),
-        startDate=format_date(p.date_from),
-        endDate=format_date(p.date_to)) for p in prices]), 200
+    return jsonify(prices=[p.serializable for p in prices]), 200
+
+
+@auth.route('/supplier/prices', methods=['POST'])
+@login_required
+@role_required('supplier')
+@swag.validate('PriceUpdates')
+def update_supplier_price():
+    """Update a price (role=supplier)
+    ---
+    tags:
+      - supplier
+    security:
+      - basicAuth: []
+    consumes:
+      - application/json
+    parameters:
+      - name: body
+        in: body
+        required: true
+        schema:
+          id: PriceUpdates
+          required:
+            - prices
+          properties:
+            prices:
+              type: array
+              items:
+                type: object
+                required:
+                  - id
+                  - price
+                  - startDate
+                additionalProperties: false
+                properties:
+                  id:
+                    type: integer
+                  price:
+                    type: number
+                  startDate:
+                    type: string
+                  endDate:
+                    type: string
+    responses:
+      200:
+        description: An updated price
+        schema:
+          properties:
+            prices:
+              type: array
+              items:
+                $ref: '#/definitions/SupplierPrice'
+    """
+    json_data = request.get_json()
+    prices = json_data.get('prices')
+    results = {}
+
+    for p in prices:
+        existing_price = db.session.query(ServiceTypePrice).get(p['id'])
+
+        if existing_price is None:
+            abort('Invalid price id: {}'.format(p['id']))
+
+        results[p['id']] = []
+        start_date = p.get('startDate')
+        end_date = p.get('endDate', None)
+
+        date_from = pendulum.parse(start_date, tz='UTC')
```

Comment:
What if the date string is invalid? https://github.com/sdispater/pendulum/blob/32c877ee3c04cba7d0efe41e5afc3908a584e52d/pendulum/parsing/__init__.py#L18

Reply:
Yes - we should handle this better